### PR TITLE
test: CF worker implements auth for composer and agents beta

### PR DESCRIPTION
**Pull request by richburdon**
Experimental CF workers for authenticated access and AI.

- [x] CF worker.
- [x] CF D1 SQL database.
- [x] admin API with API_KEY access control.
- [x] sends email on signup and authorization.
- [x] generates and tests JSX token for app auth.
- [x] AI worker demo. 

## Email Invitation Flow

https://github.com/dxos/dxos/assets/3523355/c91592f1-46d6-4a2f-b12c-7d25a0b362db

## CLI Access

https://github.com/dxos/dxos/assets/3523355/4180803e-56b3-4945-8ca7-7105b5fbaee2

### CODE REVIEW

File: packages/experimental/labs-workers/src/app/index.ts [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+      <pre>${raw(JSON.stringify(payload, undefined, 2))}</pre>
+      <a href="/reset">Logout</a>
+    </body>
+  `);
+});
+
+/**
+ * Set JWT from one time link.
+ */
+app.get('/access', async (c) => {
```
Comment (nferch) [inline packages/experimental/labs-workers/src/app/index.ts:60]: we're explicitly not supporting revocation, right?

Comment (richburdon) [reply packages/experimental/labs-workers/src/app/index.ts:60]: we could layer that in. but i don't think it's required initially.

File: packages/experimental/labs-workers/src/app/index.ts [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  // https://hono.dev/middleware/builtin/jwt
+  return jwtMiddleware(c, next);
+});
+
+/**
+ * Serve app.
+ */
+// TODO(burdon): Serve app as static resource: https://github.com/honojs/examples/tree/main/serve-static
+app.get('/', async (c) => {
+  const token = getCookie(c, 'access_token');
```
[the remark this answers is not in the document]
Comment (richburdon) [reply packages/experimental/labs-workers/src/app/index.ts:37]: i think we could store the token in HALO and embed within invitation links.
similarly, we can use the stored token to renew expired cookies.

File: packages/experimental/labs-workers/src/app/index.ts [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+const app = new Hono<Env>();
+
+/**
+ * Access control.
+ */
+app.use('/', (c, next) => {
+  const jwtMiddleware = jwt({
+    secret: c.env.JWT_SECRET,
+    cookie: 'access_token',
+  });
```
[the remark this answers is not in the document]
Comment (richburdon) [reply packages/experimental/labs-workers/src/app/index.ts:26]: i'm not sure i follow. either way, the service would require a public/private key pair?

File: packages/experimental/labs-workers/src/app/index.ts [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+  // TODO(burdon): https://hono.dev/helpers/cookie#following-the-best-practices
+  // https://stackoverflow.com/questions/37582444/jwt-vs-cookies-for-token-based-authentication
+  // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie
+  setCookie(context, 'access_token', token, {
+    // NOTE: Setting domain breaks curl.
+    secure: context.env.WORKER_ENV === 'production',
+    httpOnly: true,
+  });
+
+  // TODO(burdon): Add access token to HALO.
```
[the remark this answers is not in the document]
Comment (richburdon) [reply packages/experimental/labs-workers/src/app/index.ts:85]: there is a endpoint that can be used to validate the token, so it's the app's decision.

File: packages/experimental/labs-workers/src/app/index.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (nferch) [reply packages/experimental/labs-workers/src/app/index.ts:26]: I think I understand now per the earlier comment, the app is expected to query the validation endpoint, right?

File: packages/experimental/labs-workers/src/app/index.ts (hunk 1 above)
Comment (nferch) [reply packages/experimental/labs-workers/src/app/index.ts:60]: If we're using the validation endpoint to validate the token, any changes to the authorization will be immediately reflected there, right? Or am I misunderstanding something?

**Pull request merged**